[EV-3803] Add helm installation reference doc (#815)
tigera/docs: calico-enterprise/reference/installation/helm.mdx

---
description: Helm installation reference
---
# Helm installation reference

> You can customize the following resources and settings during {{prodname}} helm-based installation using the file, values.yaml

Reviewer: ```suggestion
You can customize the following resources and settings during {{prodname}} Helm-based installation using the file, values.yaml
```

- [Installation](api.mdx#operator.tigera.io/v1.InstallationSpec)
- [Api server](api.mdx#operator.tigera.io/v1.APIServerSpec)
- [Compliance](api.mdx#operator.tigera.io/v1.ComplianceSpec)
- [Intrusion detection](api.mdx#operator.tigera.io/v1.IntrusionDetectionSpec)
- [Log collector](api.mdx#operator.tigera.io/v1.LogCollectorSpec)
- [Log storage](api.mdx#operator.tigera.io/v1.LogStorageSpec)
- [Manager](api.mdx#operator.tigera.io/v1.ManagerSpec)
- [Monitor](api.mdx#operator.tigera.io/v1.MonitorSpec)
- [Policy recommendation](api.mdx#operator.tigera.io/v1.PolicyRecommendationSpec)
- [Authentication](api.mdx#operator.tigera.io/v1.AuthenticationSpec)
- [Application layer](api.mdx#operator.tigera.io/v1.ApplicationLayerSpec)
- [Amazon cloud integration](api.mdx#operator.tigera.io/v1.AmazonCloudIntegrationSpec)
- [Default felix configuration](../resources/felixconfig.mdx#spec)

:::note

> If you customize resources when you install {{prodname}}, the v1 apiVersion is used. However, when you customize resources

Reviewer: ```suggestion
If you customize resources when you install {{prodname}}, the `v1 apiVersion` is used. However, when you customize resources
```

> after installation (when the tigera-apiserver is running), use the v3 apiVersion.

Reviewer: ```suggestion
after installation (when the tigera-apiserver is running), use the `v3 apiVersion`.
```

:::

### Sample values.yaml

> Here is a sample`values.yaml` file with settings for custom resources. You must enable the custom resource using

Reviewer: ```suggestion
Here is a sample `values.yaml` file with settings for custom resources. You must enable the custom resource using
```

  <manager resource spec fields that user wants to configure>

monitor:
  enabled: true
  <monitor resource spec fields that user wants to configure>

compliance:
  enabled: true
  <compliance resource spec fields that user wants to configure>

policyRecommendation:
  enabled: true
  <policyRecommendation resource spec fields that user wants to configure>

authentication:
  enabled: false
  <if enabled is true: authentication resource spec fields that user wants to configure>

applicationLayer:
  enabled: false
  <if enabled is true: applicationLayer resource spec fields that user wants to configure>

amazonCloudIntegration:
  enabled: false
  <if enabled is true: amazonCloudIntegration resource spec fields that user wants to configure>

defaultFelixConfiguration:
  enabled: false
  <if enables is true: felixConfiguration resource spec fields that user wants to configure>
```

## Common customizations

Common customizations that you might want to configure are number of replicas, pod affinity, and encryption using WireGuard.

### Number of replicas
This setting defines the number of replicas for {{prodname}} components that can run simultaneously in multiple instances.

> To configure this setting, use the [controlPlaneReplicas](api.mdx#operator.tigera.io/v1.InstallationSpec).

Reviewer: ```suggestion
To configure this setting, see [controlPlaneReplicas](api.mdx#operator.tigera.io/v1.InstallationSpec).
```

The components for the replicas are:

- tigera-manager
- tigera-apiserver
- tigera-dex
- tigera-kibana
- calico-node

To set a specific replica for these components, you must provide the setting in your custom values.yaml file:
```yaml
installation:
  enable: true
  controlPlaneReplicas: 3 # desired number of replicas for the components listed above.
```

### Pod affinity

> There are different ways to define affinity for different deployments, depending on the structure. The following list of

Reviewer: Not sure what the first sentence is trying to say. Is the following correct?
```suggestion
You can define pod affinity for your deployment using any of the following Tigera components. Update the appropriate custom resource in your custom `values.yaml`.
```

Author: how about:
You can define pod affinity for the following Tigera components. Update the appropriate custom resource in your custom `values.yaml`.

tigera components can be defined for pod affinity. Update the appropriate custom resource in your custom `values.yaml`.

- tigera-apiserver: through ApiServer resource.
- calico-nodes: through CalicoNodeDaemonSet property in the Installation resource.
- calico-kubecontroller: through CalicoKubeControllersDeployment property in the Installation resource.
- compliance deployment pods (compliance-snapshotter, compliance-server, compliance-controller, compliance-benchmarker,
compliance-scaleloader, compliance-reporter): through Compliance resource.
- pods created for the Elasticsearch nodes: through LogStorage resource.

### Encryption using WireGuard

> WireGuard encryption is configured in the FelixConfiguration. To set encryption for your cluster, update the `values.yaml`file.

Reviewer: ```suggestion
[WireGuard encryption](TBD) is configured in the FelixConfiguration. To set encryption for your cluster, update the `values.yaml`file.
```

---
description: Configure Calico Enterprise resources with helm installation
---

> ## Helm Installation Reference

Reviewer: ```suggestion
## Helm installation reference
```

> You can customize your {{prodname}} installation by providing settings via `values.yaml` during helm-based installation.

Reviewer: You can customize the following resources and settings during {{prodname}} installation or after using the file, `values.yaml`.

Author: `values.yaml` is only used for installation and cannot be used after that. so maybe:
You can customize the following resources and settings during {{prodname}} helm-based installation using the file, `values.yaml`

> Your custom configurations for any of the properties for the resources listed below can be set in your custom

Reviewer: Delete this line.

`values.yaml` file.

> 

Reviewer: Links are failing in netlify. Adjust.

- [Installation](../reference/installation/api#operator.tigera.io/v1.InstallationSpec)
- [Api server](../reference/installation/api#operator.tigera.io/v1.APIServerSpec)
- [Compliance](../reference/installation/api#operator.tigera.io/v1.ComplianceSpec)
- [Intrusion detection](../reference/installation/api#operator.tigera.io/v1.IntrusionDetectionSpec)
- [Log collector](../reference/installation/api#operator.tigera.io/v1.LogCollectorSpec)
- [Log storage](../reference/installation/api#operator.tigera.io/v1.LogStorageSpec)
- [Manager](../reference/installation/api#operator.tigera.io/v1.ManagerSpec)
- [Monitor](../reference/installation/api#operator.tigera.io/v1.MonitorSpec)
- [Policy recommendation](../reference/installation/api#operator.tigera.io/v1.PolicyRecommendationSpec)
- [Authentication](../reference/installation/api#operator.tigera.io/v1.AuthenticationSpec)
- [Application layer](../reference/installation/api#operator.tigera.io/v1.ApplicationLayerSpec)
- [Amazon cloud integration](../reference/installation/api#operator.tigera.io/v1.AmazonCloudIntegrationSpec)
- [Default felix configuration](../reference/resources/felixconfig#spec)

> Any of the configs available for any of the resources above can be configured via helm installation. For a list of all

Reviewer: Delete this line.
```suggestion
Any of the configs available for any of the resources above can be configured via helm installation. For a list of all
```

available properties for each resource please refer to the [Installation reference](../reference/installation/api) page.

> You still can change / apply settings after installation to make changes to your cluster by applying any custom resource.

Reviewer: Delete this line.
```suggestion
You still can change / apply settings after installation to make changes to your cluster by applying any custom resource.
```

> Note:

Reviewer: :::note
If you customize resources when you install {{prodname}}, the `v1` apiVersion is used. However, when you customize resources after installation (when the `tigera-apiserver` is running), use the `v3` apiVersion.

:::

for any configuration that is happening at the time of the installation the `v1` apiVersion is being used. However, after installation
is successful since `tigera-apiserver` is Running, the `v3` apiVersion should be used for custom resource creations and updates.

> 

Reviewer: ## Sample values.yaml

> Here is an example `values.yaml` file for setting properties for various custom resources mentioned above. Note that,

Reviewer: ```suggestion
Here is a sample`values.yaml` file with settings for custom resources. You must enable the custom resource using `enabled: true` to provide custom configurations; custom resources set to false are ignored.
```

logCollector:
  enabled: true
  <logCollector resource spec fields that user wants to configure>

logStorage:
  enabled: true
  nodes:
    count: 1
  <logStorage resource spec fields that user wants to configure>

manager:
  enabled: true
  <manager resource spec fields that user wants to configure>

monitor:
  enabled: true
  <monitor resource spec fields that user wants to configure>

compliance:
  enabled: true
  <compliance resource spec fields that user wants to configure>

policyRecommendation:
  enabled: true
  <policyRecommendation resource spec fields that user wants to configure>

authentication:
  enabled: false
  <if enabled is true: authentication resource spec fields that user wants to configure>

applicationLayer:
  enabled: false
  <if enabled is true: applicationLayer resource spec fields that user wants to configure>

amazonCloudIntegration:
  enabled: false
  <if enabled is true: amazonCloudIntegration resource spec fields that user wants to configure>

defaultFelixConfiguration:
  enabled: false
  <if enables is true: felixConfiguration resource spec fields that user wants to configure>
```

> ## Common Customizations

Reviewer: ```suggestion
## Common customizations
```

> Some example of common customizations that you might want to configure are pod affinity, replicas, and encryption using wireguard.

Reviewer: ```suggestion
Common customizations that you might want to configure are number of replicas, pod affinity, and encryption using WireGuard.
```

### Number of replicas

> Number of replicas is configurable by the [controlPlaneReplicas](../reference/installation/api#operator.tigera.io/v1.InstallationSpec)

Reviewer: ```suggestion
This setting defines the number of replicas for {{prodname}} components that can run simultaneously in multiple instances. To configure this setting, use the [controlPlaneReplicas](../reference/installation/api#operator.tigera.io/v1.InstallationSpec). The components for the replicas are:
```

property in the installation custom resource. This property defines number of replicas for components that can run in multiple
instances simultaneously. Here is the list of these components:

- tigera-manager
- tigera-apiserver
- tigera-dex
- tigera-kibana
- calico-node

> to set a specific replica for these components you need to provide the setting in your custom values.yaml file:

Reviewer: ```suggestion
To set a specific replica for these components, you must provide the setting in your custom values.yaml file:
```

```yaml
installation:
  enable: true
  controlPlaneReplicas: 3 # desired number of replicas for the components listed above.
```

> ### Affinity

Reviewer: ```suggestion
### Pod affinity
```

> There are different ways to define affinity for different deployments depending on the structure. Here is a list of tigera components that affinity can be defined for them. Affinity for these

Reviewer: ```suggestion
There are different ways to define affinity for different deployments, depending on the structure. The following list of tigera components can be defined for pod affinity. Update the appropriate custom resource in your custom `values.yaml`.
```

components can be defined by providing configuration through the appropriate custom resource in your custom `values.yaml`
file.

- tigera-apiserver: through ApiServer resource.
- calico-nodes: through CalicoNodeDaemonSet property in the Installation resource.
- calico-kubecontroller: through CalicoKubeControllersDeployment property in the Installation resource.
- compliance deployment pods (compliance-snapshotter, compliance-server, compliance-controller, compliance-benchmarker,
compliance-scaleloader, compliance-reporter): through Compliance resource.
- pods created for the Elasticsearch nodes: through LogStorage resource.

> ### Wireguard

Reviewer: ```suggestion
### Encryption using WireGuard
```

> Wireguard settings are configurable as part of the FelixConfiguration. In order to set encryption via wireguard for your

Reviewer: ```suggestion
WireGuard encryption is configured in the FelixConfiguration. To set encryption for your cluster, update the `values.yaml`file. 
```
